src/commands/create-app.ts

```diff
@@ -182,9 +182,17 @@ const updatePackageConfig = async (options: ICreateAppOptions) => {
 };
 
 const copyTemplateFiles = async (options: ICreateAppOptions): Promise<boolean> => {
-    return copy(options.templateDirectory, options.targetDirectory, {
+    const success = await copy(options.templateDirectory, options.targetDirectory, {
         clobber: false,
     });
+
+    const indexJsPath = path.join(options.targetDirectory || '', 'index.js');
+    const indexJsContent = fs.readFileSync(indexJsPath).toString();
+
+    const indexJsContentReplaced = indexJsContent.replace("{appKey}", options.key || "app-key");
+    fs.writeFileSync(indexJsPath, indexJsContentReplaced);
```

Comment:
Should we throw in some exception handling in this function with informative feedback to the user?
Could fs.writefileSync and fs.readFileSync be converted to async operations with callbacks?

Reply:
Had the same thoughts, but since it's such a simple operation and we control the source (index.js in template) I didn't bother 🤷‍♂ We could "promisify" fs and await them, but that wouldn't change anything for the user.

Reply:
Don't need to "promisify" it's built in fs.writeFile like this:

```
fs.writeFile(indexJsPath, indexJsContentReplaced, (err) =>  {
    if(err) {
        return console.log(err);
    }
   do something...
}); 
```
But like you said, it won't really change anything for the user so it's good to go 👍

Reply:
Doing it like this would not work. You would have to wrap it in a promise and wait for it to finish (that's what promisfy does.) If we did it like you suggest it would continue from the read command to the write command without waiting for it to actually read, the same goes for write. It would continue to the next task instead of waiting for it to finish writing the file. Would have to nest the callbacks and a bunch of ugly code, or use promisify and await each I/O command

Reply:
You don't have to wrap it in a promise, fs.write file already returns a promise. You just have to await it. 
To be more specific it could be written like this:

```
await fs.writeFile(indexJsPath, indexJsContentReplaced, err =>  {
    if(err) {
        return Promise.reject(new Error('Failed to create file'));
    }
}); 
return success; 
```
Doing it this way you have more control over exceptions

Reply:
Huh. According to the documentation the fs.write/writeFile returns void, not a promise. But in newer versions of node there's a version of fs that does. E.g. import { promises as fs } from "fs"; Then we could do try { await fs.writeFile(indexJs, indexJsContentReplaced); } catch(error) { return false; }

Reply:
Ahh I see. It was written like I suggested in the initGit function 
```
const initGit = async (options: CreateAppOptions) => {
   ...
    await fs.writeFile(gitIgnorePath, "node_modules\r\n.fusion", (err: any) => {
        if (err) {
            return Promise.reject(new Error("Failed to create .gitignore file"));
        }
    });
    return;
};
```
Since it returns a void, it would just return right away instead of awaiting. 
This should maybe be rewritten then.